### DISCUSSION TITLE
Webhooks

### DISCUSSION THREAD

**Discussion by allexivan** [ctubio/Krypto-trading-bot#1120] [Q&A] [answered]
Does the bot have a webhook for placing orders?

**Comment by ctubio** [comment, marked as answer]
:koala: mmm not a webhook

the web uri available are only the following:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/bin/trading-bot/trading-bot.main.h#L29-L38

:koala: mmm but a websocket message will work

the websocket can be used with wscat (see https://github.com/ctubio/Krypto-trading-bot/issues/811 and https://github.com/ctubio/Krypto-trading-bot/issues/953), and there is a submit manual order handler.

:bulb: you have an example of the submit manual order message creation at:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/bin/trading-bot/trading-bot.client/Submit.ts#L88-L93

it just sends a json with this structure:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/lib/Krypto.ninja-client/www/js/Models.ts#L147-L153

just note that to identify messages, we add 2 characters at the beginning of the message (as a prefix to the json object without any kind of separator).

the id for submit manual order is:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/lib/Krypto.ninja-client/www/js/Models.ts#L13

so an example message could look like:
```
-p{"side":0,"price":0.69,"quantity":0.69,"timeInForce":0,"type":0}
```
(`-` is used as a POST and `=` as a GET, the letter identifies the current entity type, and the json is the entity data)

in case you wanna see where this goes, the handler is at:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/bin/trading-bot/trading-bot.data.h#L1134-L1153

that is glued at:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/bin/trading-bot/trading-bot.data.h#L2727

to thiz function:
https://github.com/ctubio/Krypto-trading-bot/blob/26c269a99ac72522a39acdba96753d3cb21bf843/src/bin/trading-bot/trading-bot.data.h#L2805-L2807

hapPy coding'¡!
